### PR TITLE
Read only events admin

### PR DESCRIPTION
**Pull request by Etenil**
Add the setting `DJANGO_EASY_AUDIT_READONLY_EVENTS` to make CRUD events read-only from the admin UI.

Setting to True will do the following:

- Disable purge
- Set CRUDEvent delete permission to false
- Set all fields of CRUDEvent to read-only

Note: _The details view of the CRUDEvent retains a "save" button, but no change can be made._

### CODE REVIEW

**Comment by jheld** [comment]
I think the change is good overall, but I do wonder if we should be taking more advantage of subclassing/different CRUDEventAdmin registration.

Some of the guided approach is good (read only is a "simple" interface), but I'd almost rather the read only stuff be as an example in the docs, rather than us trying to please everyone and possibly getting out of date as we adjust fields.

All that said, read-only as a concept for an auditing library (even from django admin) seems like a good idea to add all fields as read only by default, instead of making it an `extend`.

I do like potentially having a setting for read only events, (delete_permission etc just taking from that).

What are your thoughts from this? Am I way off?

**Comment by Etenil** [comment]
@jheld I agree that my attempt at conditionally adding read-only fields in the admin form was quite bad. I've made changes to handle the read-only fields better in the admin forms, and also to automatically pick up on all fields instead of having to maintain a list by hand.

It's cool to see that you're also interested in having this feature as it's quite core for me and the company I'm doing this for. It's a requirement for me to be able to audit what superuser do even in the event that a security breach would allow a user to elevate themselves to superuser.

I understand where you're coming from by not wanting to necessarily add this to the core library as pleasing everyone would result in bloat. From a technical standpoint it's totally doable for me to shape this in the same way as the recent custom loggers. On the other hand, I believe that many users would expect an auditing library to allow read-only event logging out of the box.

I'll defer to your judgement on this, do let me know what course of action you prefer and I'll make the relevant changes.

**Comment by Etenil** [comment]
@jheld Can I have a decision on this as I'd like to be able to move on from the branch in this PR and use the main package instead?

**Comment by jheld** [comment]
@Etenil thanks for this and the simplification (using the read only function).

I expect to release `b3` of the current version this week. If my project doesn't have issues/quick community feedback is positive, we will be able to release this or next week.

after that I am happy to merge!

**Comment by jheld** [comment]
b3 has been released, I probably won't be cutting the official release until early next week (I'm waiting to get it merged and tested on my project), but we're still planning to get this in soon after (b3 should be fine after a minimal amount of smoke testing, and then general release!).

**Comment by Etenil** [comment]
@jheld I see that the release of 1.2.2 has now been done, any idea of when this PR can be merged? Did your testing reveal any problem?

**Comment by jheld** [comment]
Hey. I'm sorry I haven't come back to this.
I have a day off tomorrow so although I cannot promise I'll be able to review again and test over the next few days, I may have some time to do so.

If you don't hear back within a week, please ping back again.

File: easyaudit/admin_helpers.py [hunk 1]
```diff
@@ -51,6 +58,11 @@ def user_link(self, obj):
     def has_add_permission(self, request, obj=None):
         return False
 
+    def has_delete_permission(self, request, obj=None):
+        if settings.READONLY_EVENTS:
```
Comment (jheld) [inline easyaudit/admin_helpers.py:62]: Minor (non-blocking):
```suggestion
        return not settings.READONLY_EVENTS
```

File: easyaudit/admin_helpers.py [hunk 2]
```diff
@@ -8,7 +8,7 @@
 except: # Django < 2.0
     from django.core.urlresolvers import reverse
 
-from django.http import HttpResponseRedirect
+from django.http import HttpResponseRedirect, HttpResponse
```
Comment (jheld) [inline easyaudit/admin_helpers.py:11]: Minor (non-blocking), but it looks like we don't need `HttpResponse` imported under the current change.

File: easyaudit/tests/test_app/tests.py [hunk 3]
```diff
@@ -157,4 +157,3 @@ def test_request_event_admin_no_users(self):
         response = self.client.get(reverse('admin:easyaudit_requestevent_changelist'))
         self.assertEqual(200, response.status_code)
         filters = self._list_filters(response.content)
-        print(filters)
```
Comment (jheld) [inline easyaudit/tests/test_app/tests.py:160]: thank you!

**Comment by jheld** [comment]
released to PyPI for `1.2.3a2`

**Pull request merged**